conductr_cli/conduct_backup.py

```diff
@@ -0,0 +1,262 @@
+import hashlib
+import json
+import logging
+import os
+import shutil
+import time
+from shutil import rmtree
+
+import requests
+from requests_toolbelt import MultipartDecoder
+
+from conductr_cli import conduct_url, conduct_request, validation
+from conductr_cli.bundle_core_info import BundleCoreInfo
+from conductr_cli.conduct_info_inspect import filter_bundles_by_id_or_name
+from conductr_cli.conduct_url import conductr_host
+from conductr_cli.exceptions import ConductBackUpError
+from conductr_cli.http import DEFAULT_HTTP_TIMEOUT
+
+
+def backup(args):
+    ConductBackup(args).backup_conductr()
+
+
+class ConductBackup(object):
+    def __init__(self, args):
+        self.args = args
+        self.bundle_id_or_name = args.bundle
+
+    def backup_conductr(self):
+        log = logging.getLogger(__name__)
+
+        backup_path, backup_name = create_backup_directory(self.args.output_path)
```

Comment:
It would be nice if the output_path could be `-` (thus stdin), or even a device (`/dev/null`). To do this, you'd build everything in a temporary directory (use `tempfile.mkdtemp()`) and then use the shazar functions to stream that to `self.args.output_path`. See `bndl_create` for an example of this pattern.